onmetal-apiserver/internal/apis/storage/validation/volume.go

```diff
@@ -100,6 +101,12 @@ func validateVolumeSpec(spec *storage.VolumeSpec, fldPath *field.Path) field.Err
 		}
 	}
 
+	if spec.Encryption != nil {
+		for _, msg := range apivalidation.NameIsDNSLabel(spec.Encryption.SecretRef.Name, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("encryption").Child("secretRef").Child("name"), spec.Encryption.SecretRef.Name, msg))
+		}
+	}
```

Comment:
should we add null validation check for secreteRef.Data encryptionKey field?

Reply:
We cant add it here as secret is not available.